testing/shuffle.cu

```diff
@@ -139,4 +135,53 @@ void TestShuffleUniformPermutation() {
   ASSERT_LESS(chi_squared, confidence_threshold);
 }
 DECLARE_VECTOR_UNITTEST(TestShuffleUniformPermutation);
+
+template <typename Vector> void TestShuffleEvenSpacingBetweenOccurances() {
+  typedef typename Vector::value_type T;
+  const uint64_t shuffle_size = 10;
```

Comment:
The bug report only had problems on sequences > 150, let's make sure we catch it with  this test.

Is there a way to set up this test was similar to work on varying sizes? I think there is a test macro for inputting various size arguments.

We might have to implement a helper function here for the chi squared threshold.

Reply:
No the bug report still had issues with smaller, they just werent as immediately obvious. For example you will notice every 5th or 6th number in the 100 length one is very close together

Reply:
We can do larger sizes but this has a factor of n^2 so we would have to significantly reduce the sample size.

Reply:
Got it. I wonder if we could come up with something linear time, not necessarily for this PR. Then we could effectively fuzz test a large range of inputs.

Another thought I had, we could test the feistel is invertible to catch any potential issues there.

Reply:
That would be a good test for the feistel, all you would have to do is test that no items are duplicated/missed and they are all within range.

As for a linear time test I can grab the test for number of times each item occurs at a given location